### ISSUE TITLE
App Service Auth v2 token store enabled always plans false -> true

### ISSUE THREAD

**Issue by collins-benj** [hashicorp/terraform-provider-azurerm#21066]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.2

### AzureRM Provider Version

3.48.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_function_app, azurerm_windows_web_app

### Terraform Configuration Files

```hcl
auth_settings_v2 {

    login {
      token_store_enabled = true

      # TODO: create azurerm github issue to clarify if the tf default of 72 or the azure default of 0 is correct
      # currently we can't have 0 here
      token_refresh_extension_time = 1
    }

    default_provider = "azureactivedirectory"
    auth_enabled = true
    require_authentication = true
    runtime_version = "~1"

    active_directory_v2 {
      client_id = azuread_application.main.application_id
      allowed_audiences = []
      tenant_auth_endpoint = "https://login.microsoftonline.com/<actuall tenant redacted>/v2.0"
      client_secret_setting_name = "MICROSOFT_PROVIDER_AUTHENTICATION_SECRET"

      login_parameters = {
         # see https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow#send-the-sign-in-request for options
      prompt = "select_account"

      # offline_access allows getting of refresh token
      # see https://docs.microsoft.com/en-us/azure/app-service/configure-authentication-oauth-tokens#refresh-auth-tokens
      scope = "openid profile email offline_access"
      }
    }
  }
```

### Debug Output/Panic Output

```shell
none
```

### Expected Behaviour

after apply with `token_store_enabled = true`, `terraform plan` should be clean, with no planned changes

### Actual Behaviour

apply does set `token_store_enabled = true` properly, through [Azure Resource Explorer](https://resources.azure.com/), navigating to `authsettingsV2` shows the following: 
![image](https://user-images.githubusercontent.com/97915450/226719690-f8cc8902-dcb8-4133-a054-95b9a7b4e42a.png)
yet the terraform plan outputs

```
~ auth_settings_v2 {
            # (9 unchanged attributes hidden)

          ~ login {
              ~ token_store_enabled               = false -> true
```

applying again at this stage appears to do nothing.

### References

this bug is also doc'd in comments of #21021

**Comment by xiaxyi** [comment]
@collins-benj Thanks for raising this issue, the isue should be fixed by PR https://github.com/hashicorp/terraform-provider-azurerm/pull/21091. 

Can you help me to test it and let me know if you are still seeing any issues?

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #21091. As such, I am going to mark this issue as closed.
